orangecrab/src/Packet.hs

```diff
@@ -57,27 +53,23 @@ instance IlaPacketType IlaDataHeader where
 
 -- | Construct a data packet from a stream of raw data
 dataPacket ::
-  forall dom dataWidth size t.
+  forall dom dataWidth size.
   ( HiddenClockResetEnable dom
   , KnownNat dataWidth
   , KnownNat size
-  , BitPack t
   , 1 <= dataWidth
   , 1 <= size
   ) =>
-  Proxy t ->
   -- | Circuit which takes in a datastream with the length as metadata and outputs packaged data
   Circuit
-    (PacketStream dom dataWidth (BitVector 16, Index size))
+    (PacketStream dom dataWidth (BitVector 32, Index size))
```

Comment:
I understand that here the hash is part of packetstreams metadata, right?
But is this not compile time known? I think it should be a normal (non-signal) argument to your `dataPacket` function

Reply:
Furthermore you don't seem to use the other part of the meta's tuple, I think the meta should be `()`